### ISSUE TITLE
Rayman Origins (207490)

### ISSUE THREAD

**Issue by pprotonrep** [ValveSoftware/Proton#43]
I tested [Rayman Origins](https://store.steampowered.com/app/207490/) and it worked perfectly fine for me, including sounds, controller input and effects.

- Ubuntu 18.04.1 LTS (64 bit 4.15.0-32-generic)
- GeForce GTX 1070/PCIe/SSE2 (396.51)
- Proton 3.7 (default configuration)

[System information from steam](https://gist.github.com/pprotonrep/43f7265a7457b17ceb5b848e89c26147)

**Comment by Cuddlypuff** [comment]
same here, completed first level without any issues
https://gist.github.com/Cuddlypuff/a518019d119623d9a1b7552e284c1f6d

**Comment by KRFTSPP** [comment]
Most of the game works fine for me, _except_ a gameplay breaking bug which renders some of the platform sprites invisible in later levels of the game. One of the earlier examples of this bug is on the level "Over the rainbow" in "Jibberish Jungle".

There's also a bug report on WineHQ about this: [https://bugs.winehq.org/show_bug.cgi?id=30550](https://bugs.winehq.org/show_bug.cgi?id=30550)

I think Rayman Origins shouldn't be whitelisted before this bug is solved.

My hardware:

- Arch Linux (4.18.5.arch1-1)
- Asus NVIDIA GTX 1050 Ti 4GB (396.54)
- Proton 3.7-4 Beta

https://gist.github.com/KRFTSPP/303de1183dbc7175ba7535dc93088304

**Comment by Tyler-2** [comment]
Same issue here. "My Heartburn's For You" fire is invisible rendering the level impossible. Also the overview where you pick worlds is missing some paths on the completed world.

**Comment by kisak-valve** [comment]
Hello @Tyler-2, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

For those using nVidia, please retest with Proton 3.16 and nvidia 415.22 or newer.

**Comment by KRFTSPP** [comment]
I just tried it out and the issue still persists for me on the same level as last time - "Over the rainbow" in "Jibberish Jungle". I'm currently on Proton 3.16-6 Beta and Nvidia 415.27.

https://gist.github.com/KRFTSPP/0fdf0a39c4574090608093a232d46f01

**Comment by kisak-valve** [comment]
Invisible sprites in this game is a long standing issue being tracked upstream at https://bugs.winehq.org/show_bug.cgi?id=30550.

**Comment by rdotts** [comment]
Ran into this myself: the game was running great until I started to die from invisible tentacles on one level.  I managed to find some threads with complaints about the issue from 2012, so it looks like this is a Rayman Origins bug, not a Proton bug.

https://forums.ubi.com/showthread.php/663042-Rayman-Origins-Pc-graphical-problem

You can find a few other examples of people complaining by googling "Rayman Origins bug invisible".  There's also a few who also said it resolved itself by playing with resolutions or reinstalling entirely, but I couldn't find anything people said worked consistently.

**Comment by shoober420** [comment]
# Whitelist Request
- Name of the game to be whitelisted: Rayman Origins
- Steam AppID of the game: https://steamdb.info/app/207490/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.59
- Distro version: Arch
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-3

Works great out of the box.

**Comment by kisak-valve** [comment]
Hello is anyone experiencing an issue with this game and Proton 5.0 that would block it from being whitelisted? In particular, has DXVK/D3D9, which is enabled by default in this Proton release, resolved the previous issues with invisible objects?

**Comment by Tyler-2** [comment]
I'll install it again and check.

**Comment by Tyler-2** [comment]
I don't have a save-game for the parts that caused me the most trouble (fire in some level), but I just played through the first level and no problems so far.

**Comment by yajo** [comment]
1-2 weeks ago I bought the game and I played it with zero problems using Proton 5.13 (right now it's 5.13-4, but I'm not sure if it got any updates during this time).

Now it doesn't work. Also not in other older proton versions I tried. Only on proton experimental. What happened? How can I help debug?

**Comment by kisak-valve** [comment]
Hello @Yajo, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by yajo** [comment]
All info here:

1. [steam-207490.log](https://github.com/ValveSoftware/Proton/files/5760629/steam-207490.log)
1. [system info.txt](https://github.com/ValveSoftware/Proton/files/5760633/system.info.txt)

**Comment by yajo** [comment]
BTW I think it will help to clarify what it means "it doesn't work".

The game starts. I can see just about 1-2s of the 1st intro video. It goes smooth and has sound. But after those 1-2 seconds, it dies.

**Comment by kisak-valve** [comment]
Thanks @Yajo, I took a look and unfortunately I'm not seeing a strong hint in the log and there hasn't been many changes in the last couple weeks due to the holidays.

Just to confirm, the game works for you with Proton Experimental, but no other Proton build? It looks like you're using an nVidia Optimus-based laptop. Have you changed any Optimus-related config since you last saw the game working fine?

Maybe try running the game with the Intel GPU and see if that has an effect.

**Comment by yajo** [comment]
> Just to confirm, the game works for you with Proton Experimental, but no other Proton build?

Yes, and there it shows some artifacts on some levels. I have not tried *all* proton versions, but I tried the last 4 available.



> It looks like you're using an nVidia Optimus-based laptop. Have you changed any Optimus-related config since you last saw the game working fine?

Not that I know... Maybe a system upgrade.



> Maybe try running the game with the Intel GPU and see if that has an effect.

How to do it?

**Comment by friedc** [comment]
Xbox 360 controller used to work but no longer does. I am using proton-7.0-4-rc3 [log](https://github.com/ValveSoftware/Proton/files/9284942/steam-207490.log).
Controller works with proton-5.0-10b [log](https://github.com/ValveSoftware/Proton/files/9284975/steam-207490.log)

**Comment by friedc** [comment]
Xbox 360 controller works with proton-7.0-5-rc4.
